helix-core/src/main/java/org/apache/helix/model/Message.java

```diff
@@ -937,6 +946,13 @@ private boolean isNullOrEmpty(String data) {
     return data == null || data.length() == 0 || data.trim().length() == 0;
   }
 
+  private void initStat() {
```

Comment:
This looks ugly.

Let's do this,

  public Message(ZNRecord record, String id) {
    _record = new SessionAwareZNRecord(record, id);
    _stat = new Stat(_record.getVersion(), _record.getCreationTime(), _record.getModifiedTime(),
        _record.getEphemeralOwner());
  }

Then the other constructors refer to this one.

Reply:
Actually the existing code is already messing :(